File: goat_latin.pl
#!/usr/bin/perl

use strict;
use warnings;

my $string = 'I speak a bit of Goat Latin';

sub convert_to_goat_latin {
    my $str = shift;
    my @array = split / /, $str;
    my @new_array;

    my $counter = 0;
    for ( @array ) {
        if ( $_ =~ /^[aeiouy]/i ) {
            my $vowel_string = $_.add_some_as( $counter );
            push @new_array, $vowel_string;
        } else {
            my $first_char = substr( $_, 0, 1 ); 
            substr( $_, 0, 1, ""); 
            my $consonant_string = $_.$first_char.add_some_as( $counter ); 
            push @new_array, $consonant_string; 
        }
        $counter++;
    }
    my $gl_string = join( " ", @new_array ); 
    return $gl_string;
}

sub add_some_as {
    my $quantity = shift;
    return "a" x $quantity;
}

my $newstring = convert_to_goat_latin( $string ); 
print $newstring, "\n"; 

File: goat_latin.py
#!/usr/bin/env python

def convert_to_goat_latin( str ):
    out = []
    words = str.split()
    counter = 0
    for word in words:
        if word[0] == "aeiouy":
            out.append( ( word[1:] ) + ( "a" * counter ) )
        else:
            first_char = word[0:1]
            out.append(( word[1:] ) + first_char + ( "a" * counter ) )
        counter += 1
    return out

string = 'I speak a bit of Goat Latin'
print ' '.join( convert_to_goat_latin( string ) )
